requirements.txt

```diff
@@ -17,13 +17,10 @@
 # pip-install this one as editable using this repository's setup.py file.  The
 # same modules should appear in the setup.py list as given below.
 
-black==21.5b1
-filelock==3.0.12
-# Pin importlib-metadata to <5 due to https://github.com/python/importlib_metadata/issues/409.
-importlib-metadata==4.13.0
-pytest==6.2.4
-pytest-xdist==2.3.0
-filelock==3.0.12
+black
+filelock
+pytest
+pytest-xdist
```

Comment:
Not sure why these were unpinned - if newer versions are needed for Python 3.12 we should pin to the latest.